ext/phar/util.c

```diff
@@ -1006,42 +1007,41 @@ zend_result phar_get_archive(phar_archive_data **archive, char *fname, size_t fn
 
 	if (PHAR_G(last_phar) && fname_len == PHAR_G(last_phar_name_len) && !memcmp(fname, PHAR_G(last_phar_name), fname_len)) {
 		*archive = PHAR_G(last_phar);
-		if (alias && alias_len) {
+		if (alias) {
 
-			if (!PHAR_G(last_phar)->is_temporary_alias && (alias_len != PHAR_G(last_phar)->alias_len || memcmp(PHAR_G(last_phar)->alias, alias, alias_len))) {
+			if (!PHAR_G(last_phar)->is_temporary_alias && !zend_string_equals(PHAR_G(last_phar)->alias, alias)) {
 				if (error) {
-					spprintf(error, 0, "alias \"%s\" is already used for archive \"%s\" cannot be overloaded with \"%s\"", alias, PHAR_G(last_phar)->fname, fname);
+					spprintf(error, 0, "alias \"%s\" is already used for archive \"%s\" cannot be overloaded with \"%s\"", ZSTR_VAL(alias), PHAR_G(last_phar)->fname, fname);
 				}
 				*archive = NULL;
 				return FAILURE;
 			}
 
-			if (PHAR_G(last_phar)->alias_len && NULL != (fd_ptr = zend_hash_str_find_ptr(&(PHAR_G(phar_alias_map)), PHAR_G(last_phar)->alias, PHAR_G(last_phar)->alias_len))) {
-				zend_hash_str_del(&(PHAR_G(phar_alias_map)), PHAR_G(last_phar)->alias, PHAR_G(last_phar)->alias_len);
+			if (PHAR_G(last_phar)->alias && NULL != (fd_ptr = zend_hash_find_ptr(&(PHAR_G(phar_alias_map)), PHAR_G(last_phar)->alias))) {
+				zend_hash_del(&(PHAR_G(phar_alias_map)), PHAR_G(last_phar)->alias);
 			}
```

Comment:
Similarly, can't we just `hash_del` unconditionally here?

Reply:
Not here because we need fd_ptr